compaction/compaction_manager.cc

```diff
@@ -396,6 +395,11 @@ void compaction_manager::task::stop(sstring reason) noexcept {
     compaction_data.stop(std::move(reason));
 }
 
+void compaction_manager::task::throw_compaction_stopped_exception() const {
+    auto s = compacting_table->schema();
+    throw sstables::compaction_stopped_exception(s->ks_name(), s->cf_name(), compaction_data.stop_requested);
```

Comment:
maybe it will be cleaner to return the exception and let the caller throw it
(or it can return it as exceptional future using `make_exception_ptr`)

Reply:
good idea. exceptions are heavy weight, better to avoid them with make_exception_ptr.